server/src/main/scala/za/co/absa/atum/server/api/controller/PartitioningControllerImpl.scala

```diff
@@ -16,39 +16,53 @@
 
 package za.co.absa.atum.server.api.controller
 
-
 import za.co.absa.atum.model.dto.{AdditionalDataSubmitDTO, AtumContextDTO, PartitioningSubmitDTO}
 import za.co.absa.atum.server.api.exception.ServiceError
 import za.co.absa.atum.server.api.service.PartitioningService
-import za.co.absa.atum.server.model.{ErrorResponse, InternalServerErrorResponse}
+import za.co.absa.atum.server.model.ErrorResponse.{ErrorResponse, InternalServerErrorResponse}
+import za.co.absa.atum.server.model.SuccessResponse.SingleSuccessResponse
 import zio._
 
 class PartitioningControllerImpl(partitioningService: PartitioningService)
-  extends PartitioningController with BaseController {
+    extends PartitioningController
+    with BaseController {
 
-  override def createPartitioningIfNotExists(
+  override def createPartitioningIfNotExistsV1(
     partitioningSubmitDTO: PartitioningSubmitDTO
-      ): IO[ErrorResponse, AtumContextDTO] = {
-    for {
-      _ <- partitioningService.createPartitioningIfNotExists(partitioningSubmitDTO)
+  ): IO[ErrorResponse, AtumContextDTO] = {
+    val atumContextDTOEffect = for {
+      _ <- partitioningService
+        .createPartitioningIfNotExists(partitioningSubmitDTO)
         .mapError(serviceError => InternalServerErrorResponse(serviceError.message))
-      measures <- partitioningService.getPartitioningMeasures(partitioningSubmitDTO.partitioning)
-        .mapError {
-          serviceError: ServiceError => InternalServerErrorResponse(serviceError.message)
-      }
-      additionalData <- partitioningService.getPartitioningAdditionalData(partitioningSubmitDTO.partitioning)
-        .mapError {
-          serviceError: ServiceError => InternalServerErrorResponse(serviceError.message)
-      }
+      measures <- partitioningService
+        .getPartitioningMeasures(partitioningSubmitDTO.partitioning)
+        .mapError { serviceError: ServiceError =>
+          InternalServerErrorResponse(serviceError.message)
+        }
+      additionalData <- partitioningService
+        .getPartitioningAdditionalData(partitioningSubmitDTO.partitioning)
+        .mapError { serviceError: ServiceError =>
+          InternalServerErrorResponse(serviceError.message)
+        }
     } yield AtumContextDTO(partitioningSubmitDTO.partitioning, measures.toSet, additionalData)
+
+    atumContextDTOEffect
+  }
+
+  override def createPartitioningIfNotExistsV2(
+    partitioningSubmitDTO: PartitioningSubmitDTO
+  ): IO[ErrorResponse, SingleSuccessResponse[AtumContextDTO]] = {
+    mapToSingleSuccessResponse(createPartitioningIfNotExistsV1(partitioningSubmitDTO))
   }
 
   override def createOrUpdateAdditionalData(
```

Comment:
I'm wondering whether we should rename these to `v2` already. Because now it's:
* v1
* v2
* none (but they represent v2 basically)

What do you think?

Reply:
renamed to createOrUpdateAdditionalDataV2